google/cloud/vmwareengine/BUILD.bazel

```diff
@@ -0,0 +1,75 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+package(default_visibility = ["//visibility:private"])
+
+licenses(["notice"])  # Apache 2.0
+
+service_dirs = [
+    "v1/",
+]
+
+internal_dirs = [
+    "",
+    "internal/",
+]
+
+src_dirs = [s + i for s in service_dirs for i in internal_dirs]
```

Comment:
No action required: I think in almost all cases it will be easier to just unroll these lists.  Most services will have 1 or 2 versions, so we can just say:

```
service_dirs=["v1/"]
srcs_dirs = ["v1/", "v1/internal/"]
```

or (in a few cases):

```
service_dirs=["v1/", "v2/"]
srcs_dirs = ["v1/", "v1/internal/", "v2/", "v2/internal/"]
```

That is easier to grok and maintain that the for loops (I think).  If we must use for loops we should consider:

```
service_dirs = ["v1/", "v2/"]
src_dirs = service_dirs + [i + "internal/" for i in service_dirs]
```

Reply:
I believe I concur. I'll chat to @dbolduc about it, and see what can be done under cover of https://github.com/googleapis/google-cloud-cpp/issues/10170.

Reply:
It's not a question about how many versions. It is a question about how many subdirectories.

I concede that the most common size of the list will probably be 2. (if/when we move the services to a new dir and leave backwards compat headers). The worst case (that I know of) is `bigquery` which will need:

```
set(service_dirs "" "analyticshub/v1/" "connection/v1/"
                 "datatransfer/v1/" "migration/v2/"
                 "reservation/v1/" "storage/v1/" "v2/")
```

The crazy plan I am considering is to
1. #10237
2. factor out the `cc_library(...)`'s into a `google/cloud/foo/foo.bzl`. (I think we only need the `deps`. There are libraries that are [only compatible on certain platforms](https://github.com/googleapis/google-cloud-cpp/blob/7d35ce8ea8a14d66c7a1f64acd7d36eef3812626/google/cloud/asset/BUILD.bazel#L44-L48). I hope this could be captured in the generator configuration. Which would also generate [documentation](https://github.com/googleapis/google-cloud-cpp/blame/main/google/cloud/asset/README.md#L11-L17) that is not stale.)
3. generate the `BUILD.bazel` file. (we know all the subdirectories; they are defined in the generator configuration).

The `defs.bzl` would be initially created by the scaffold generator, and maintained by hand. We might do a similar thing with `CMakeLists.txt` -> `lib.cmake` for the quickstart configuration.

Am I actively working on this crazy plan? No. It is just a thought that I would explore if I had infinite time.